scipy/optimize/_optimize.py

```diff
@@ -3469,6 +3472,15 @@ def brute(func, ranges, args=(), Ns=20, full_output=0, finish=fmin,
     gridpoint [-1.0 1.75] where the rounded function value is -2.892.
 
     """
+    if finish and (integrality is not None) and np.any(integrality):
```

Comment:
We could write it that way, but why is the nested `if` better?

Reply:
There's no real need to create an `integrality` array unless the user wants to use that feature, that's why I suggested putting it in an if-clause. If that happens, then one can place the check for polishing in there as well.

Reply:
K. Avoided creating an `integrality` array unless it's entered by the user. Not my preferred way, because it adds extra conditions and sometimes extra indentation throughout the rest of the code, but we'll try it out.